tests/cpgrid/geometry_test.cpp

```diff
@@ -237,10 +237,142 @@ BOOST_AUTO_TEST_CASE(cellgeom)
             BOOST_CHECK_EQUAL(g.jacobianInverseTransposed(testpts[i]), Jit);
         }
     }
+}
+
+
+void
+check_refined_grid(const cpgrid::Geometry<3, 3>& parent,
+                   const std::vector<cpgrid::Geometry<3, 3>>& refined,
+                   const std::array<int, 3>& cells)
+{
+    using Geometry = cpgrid::Geometry<3, 3>;
+    using GlobalCoordinate = Geometry::GlobalCoordinate;
+
+    int count = cells[0] * cells[1] * cells[2];
+    BOOST_CHECK_EQUAL(refined.size(), count);
+
+    // Check the position of the first refined cell.
+    for (int c = 0; c < 3; c++) {
+        BOOST_CHECK_CLOSE(refined[0].corner(0)[c], parent.corner(0)[c], 1e-6);
+    }
+    // Check the last corner of the last cell.
+    for (int c = 0; c < 3; c++) {
+        BOOST_CHECK_CLOSE(refined[count - 1].corner(7)[c], parent.corner(7)[c], 1e-6);
+    }
```

Comment:
We might want to check more corners. Should be 8 that coincide,

Reply:
I will do that, this is indeed a bit minimal...

Reply:
Seems resolved